### PR TITLE
Assign error code when concatenation fails

### PR DESCRIPTION
**Pull request by pawelduda**
*Issue #27, if available:*

*Description of changes:*
Solves issue #27

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by pawelduda** [comment]
Feel free to give me some feedback, I am not a Kotlin person at all

File: lang/src/org/partiql/lang/eval/ExprValueExtensions.kt [hunk 1]
```diff
@@ -77,8 +77,11 @@ fun ExprValue.numberValue(): Number =
 fun ExprValue.timestampValue(): Timestamp =
     scalar.timestampValue() ?: errNoContext("Expected timestamp: $ionValue", internal = false)
 
-fun ExprValue.stringValue(): String =
-    scalar.stringValue() ?: errNoContext("Expected text: $ionValue", internal = false)
+fun ExprValue.stringValue(): String = scalar.stringValue() ?:
```
Comment (therapon) [inline lang/src/org/partiql/lang/eval/ExprValueExtensions.kt]: This is a general extension function that is called in multiple use cases that require to obtain a `String` from an `ExprValue`, one such case is when we need to evaluate a string append `||` operator. 

Making this extension function report an error that *always* states it is due to a string append operation might lead to miscommunication. 

I think the logic needs to go [here](https://github.com/partiql/partiql-lang-kotlin/blob/1ad2e93c1192d79029f22f687587d362eb328454/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L516) where the code deals with string append. Each argument to `||` needs to be checked and if it is not a `String` value then report the more specific error message that `||` did not receive a value of the appropriate type (`String`)

Comment (pawelduda) [reply lang/src/org/partiql/lang/eval/ExprValueExtensions.kt]: Makes sense - I'll attempt to fix it before next week.

Comment (pawelduda) [reply lang/src/org/partiql/lang/eval/ExprValueExtensions.kt]: @therapon Can anyone take this over? A lot going on in my life, I would love to help you out but I cannot do that. Should I just trash this PR?

Comment (pawelduda) [reply lang/src/org/partiql/lang/eval/ExprValueExtensions.kt]: No problem, I'm glad I could contribute something meaningful.

**Comment by raganhan** [comment]
https://github.com/partiql/partiql-lang-kotlin/pull/131 continuing this work

**Comment by raganhan** [comment]
https://github.com/partiql/partiql-lang-kotlin/pull/131 was pushed

Thanks @pawelduda for your contribution :)

**Pull request closed without merging**